.gitignore

```diff
@@ -30,7 +30,6 @@ build/Release
 node_modules
 jspm_packages
 
-package-lock.json
```

Comment:
don't commit package-log file, others LGTM

Reply:
@dead-horse how come? `package-lock.json` is intended to be committed to source control

Reply:
package-lock file will kill semver minor and patch upgrades and add burden on maintainers to keep package updated.

Imagine having a ranged dependency, such as ^1.2.3 (Which is >=1.2.3 <2.0.0). The package has a flaw and releases a patch, instead of having maintainers having to update the lock-file, users can simply re-install or `npm update`. lock-files messes with `npm audit fix`.

Reply:
@fl0w thanks for that. I'll remove `package-lock.json` tonight

Reply:
@fl0w @dead-horse This is a common misconception, but it's incorrect -- [lockfiles are never published to npm](https://docs.npmjs.com/files/package-lock.json), and during installs, they're only examined at the top level of a project. They never prevent library consumers from upgrading transitive dependencies. This is also true of `yarn.lock`, [which is always recommended to be checked into source control](https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/#libraries-vs-applications). (I strongly recommend reading that article for a comprehensive treatment of the issue.)

@divmgl was correct in the first place, and `package.lock.json` should be checked in to this repo. It'll make things faster, more reproducible and more reliable for developers without any negative impact on consumers of the library.

Reply:
@divmgl I agree with @lostfictions that the package-lock needs to be checked in, you can implicitly ignore a file for npm by adding a  .npmignore file and add that in there. 
source: https://zellwk.com/blog/ignoring-files-from-npm-package/

Reply:
Okay, can we get this merged in @dead-horse ?

Reply:
there is both pros and cons for the lockfile and I don't want to argue here. but this project won't accept commit lockfile.

Reply:
@dead-horse why won't it accept the lockfile?

Reply:
I for one am in favour of NOT including package-lock.json since regular/cron CI will not pickup dependency and transitive dependency new versions and warn on potential breaking changes in those.